src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala

```diff
@@ -592,6 +603,54 @@ object TezosDatabaseOperations extends LazyLogging {
       .getOrElse(defaultBlockLevel.toInt)
       .result
 
+  /** Get next batch of voting levels */
+  def fetchVotingBlockLevels(implicit ec: ExecutionContext): DBIO[Set[Int]] = {
+
+    def fetchVotesMaxLevel: DBIO[Int] =
+      Tables.Votes
+        .map(_.level)
+        .max
+        .getOrElse(defaultBlockLevel.toInt)
+        .result
+
+    for {
+      headBlockLevel <- fetchMaxBlockLevel
+      voteMaxLevel <- fetchVotesMaxLevel
+    } yield (voteMaxLevel + 1 to headBlockLevel).toSet
+  }
+
+  /** Voting data representation */
+  case class VotingData(
+      proposalHash: Option[String],
+      level: Int,
+      hash: String,
+      cycle: Option[Int],
+      timestamp: java.sql.Timestamp,
+      ballot: Option[String],
+      address: Option[String]
+  )
+
+  /** Get necessary fields to populate votes table from ballot operations */
+  def fetchVotingData(levels: Set[Int])(implicit ec: ExecutionContext): DBIO[Seq[VotingData]] =
+    Tables.Operations
+      .filter(_.kind inSet Set("ballot"))
+      .filter(_.blockLevel inSet levels)
+      .result
```

Comment:
I'm not sure it will perform better, but I think we can directly map to the fields of interest (project the query result) and then use slick's api to get directly the VotingData objects in the `.result` outcome

Reply:
I picked the columns I needed into columns, but I couldn't map them into a case class before the result because those were `Rep[]`, I changed it a bit, tell me if it's ok with you

Reply:
I guess the idea was that you can call `mapTo[VotingData]` as the last operation using slick support for case classes

see https://scala-slick.org/doc/3.3.0/gettingstarted.html#case-class-mapping

Reply:
but from what I understand you'll need to define the mapping beforehand, right?

Reply:
I think you should be able to call that right after the second `.filter`? But I'm not totally sure with that.

Reply:
either we have something different in mind or it does not work the way you're suggesting

Reply:
It's entirely possible that I'm wrong, so you can leave it at that if it doesn't work.